dynamic_programming/longest_common_subsequence.py

```diff
@@ -1,37 +1,83 @@
 """
 LCS Problem Statement: Given two sequences, find the length of longest subsequence present in both of them.
-A subsequence is a sequence that appears in the same relative order, but not necessarily continious.
+A subsequence is a sequence that appears in the same relative order, but not necessarily continuous.
 Example:"abc", "abg" are subsequences of "abcdefgh".
 """
 from __future__ import print_function
 
-try:
-    xrange          # Python 2
-except NameError:
-    xrange = range  # Python 3
 
-def lcs_dp(x, y):
+def lcs_dp(x: str, y: str):
```

Comment:
Should we have a more intuitive function name?

I would just return the substring because the caller can easily do __len(substring)__ if that information is really needed.